### PR TITLE
domain: support plan replayer dump workers running in concurrency

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38779

Problem Summary:

### What is changed and how it works?

 support plan replayer dump workers running in concurrency

### Check List

Tests 

- [x] Unit test
- [x] Manual test (add detailed scripts or steps below)

https://pingcap.feishu.cn/docx/Ak9WdC71No5PybxifvocUK59nRX?from=from_lark_index_search

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -655,14 +655,15 @@ type Performance struct {
...
-	AnalyzePartitionConcurrencyQuota uint   `toml:"analyze-partition-concurrency-quota" json:"analyze-partition-concurrency-quota"`
-	EnableStatsCacheMemQuota         bool   `toml:"enable-stats-cache-mem-quota" json:"enable-stats-cache-mem-quota"`
+	IndexUsageSyncLease               string `toml:"index-usage-sync-lease" json:"index-usage-sync-lease"`
+	PlanReplayerGCLease               string `toml:"plan-replayer-gc-lease" json:"plan-replayer-gc-lease"`
+	GOGC                              int    `toml:"gogc" json:"gogc"`
+	EnforceMPP                        bool   `toml:"enforce-mpp" json:"enforce-mpp"`
+	StatsLoadConcurrency              uint   `toml:"stats-load-concurrency" json:"stats-load-concurrency"`
+	StatsLoadQueueSize                uint   `toml:"stats-load-queue-size" json:"stats-load-queue-size"`
+	AnalyzePartitionConcurrencyQuota  uint   `toml:"analyze-partition-concurrency-quota" json:"analyze-partition-concurrency-quota"`
+	PlanReplayerDumpWorkerConcurrency uint   `toml:"plan-replayer-dump-worker-concurrency" json:"plan-replayer-dump-worker-concurrency"`
```
Comment (qw4990) [inline config/config.go:665]: How about using a SQL variable instead of a config item? SQL variables are much easier to use.

Comment (Yisaer) [reply config/config.go:665]: we can't adjust worker count in runtime thus we use config.

File: domain/domain.go [hunk 2]
```diff
@@ -1635,14 +1650,11 @@ func (do *Domain) StartPlanReplayerHandle() {
...
-			case <-do.exit:
-				return
-			case task := <-do.planReplayerHandle.planReplayerTaskDumpHandle.taskCH:
-				do.planReplayerHandle.HandlePlanReplayerDumpTask(task)
-			}
+		for _, worker := range do.planReplayerHandle.planReplayerTaskDumpHandle.workers {
+			go worker.run()
 		}
+		<-do.exit
+		do.planReplayerHandle.planReplayerTaskDumpHandle.Close()
```
Comment (xuyifangreeneyes) [inline domain/domain.go:1657]: Do we need to also put `Close` in defer?

Comment (Yisaer) [reply domain/domain.go]: updated.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**